core/payments/vault/payment_vault.go

```diff
@@ -0,0 +1,91 @@
+package vault
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/Layr-Labs/eigenda/common"
+	bindings "github.com/Layr-Labs/eigenda/contracts/bindings/PaymentVault"
+	"github.com/Layr-Labs/eigenda/core/payments"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+// Provides access to PaymentVault contract
+type paymentVault struct {
+	logger              logging.Logger
+	paymentVaultBinding *bindings.ContractPaymentVault
+}
+
+var _ payments.PaymentVault = &paymentVault{}
+
+func NewPaymentVault(
+	logger logging.Logger,
+	ethClient common.EthClient,
+	paymentVaultAddress gethcommon.Address,
+) (payments.PaymentVault, error) {
+	if ethClient == nil {
+		return nil, errors.New("ethClient cannot be nil")
+	}
+
+	paymentVaultBinding, err := bindings.NewContractPaymentVault(paymentVaultAddress, ethClient)
+	if err != nil {
+		return nil, fmt.Errorf("new contract payment vault: %w", err)
+	}
+
+	return &paymentVault{
+		logger:              logger,
+		paymentVaultBinding: paymentVaultBinding,
+	}, nil
+}
+
+// Retrieves total deposit information for multiple accounts
+func (pv *paymentVault) GetTotalDeposits(
+	ctx context.Context,
+	accountIDs []gethcommon.Address,
+) ([]*big.Int, error) {
+	totalDeposits, err := pv.paymentVaultBinding.GetOnDemandTotalDeposits(&bind.CallOpts{Context: ctx}, accountIDs)
+	if err != nil {
+		return nil, fmt.Errorf("get on demand total deposits eth call: %w", err)
+	}
+	return totalDeposits, nil
+}
+
+// Retrieves total deposit information for a single account
+func (pv *paymentVault) GetTotalDeposit(ctx context.Context, accountID gethcommon.Address) (*big.Int, error) {
+	onDemandPayment, err := pv.paymentVaultBinding.GetOnDemandTotalDeposit(&bind.CallOpts{Context: ctx}, accountID)
+	if err != nil {
+		return nil, fmt.Errorf("get on demand total deposit for account %v eth call: %w", accountID.Hex(), err)
+	}
+	return onDemandPayment, nil
+}
+
+// Retrieves the global symbols per second parameter
+func (pv *paymentVault) GetGlobalSymbolsPerSecond(ctx context.Context) (uint64, error) {
+	globalSymbolsPerSecond, err := pv.paymentVaultBinding.GlobalSymbolsPerPeriod(&bind.CallOpts{Context: ctx})
```

Comment:
I feel like we should start anchoring all our eth calls in RBN. Is that possible? Otherwise we'll get inconsistent results if/when we change a value on-chain. Whether we care about these small inconsistencies is another questino though.